v2g-liberty/rootfs/root/appdaemon/apps/v2g_liberty/reservations_client.py

```diff
@@ -502,9 +502,8 @@ async def __process_v2g_events(self, new_v2g_events):
 
     async def __set_events_in_main_app(self, v2g_args: str = ""):
         try:
-            await self.v2g_main_app.handle_calendar_change(
-                v2g_events=self.v2g_events, v2g_args=v2g_args
-            )
+            self.emit("calendar_change", v2g_events=self.v2g_events, v2g_args=v2g_args)
+            await self.wait_for_complete()
```

Comment:
Since the callback is `async`, I am using the async event emitter (even though I say this would not be necessary...).
To ensure the async method is done, we need to call this after emitting the event.

(I would have preferred a `await async_emit(...)` instead on the library.)

Reply:
Seems a bit unusual to me too. Any idea what the impact might be? Doc says this is intended for graceful shutdowns?

Reply:
I do not understand what shutdown has to do with this.
It is just a matter of API choice. Apparently they wanted to keep using "emit" real bad and did not want to change the semantics.